packages/svelte/src/lib/component/UploadButton.svelte

```diff
@@ -0,0 +1,263 @@
+<script lang="ts" context="module">
+  // Workaround for eslint-plugin-svelte issue regarding generics:
+  // https://github.com/sveltejs/svelte-eslint-parser/issues/306
+  import type { FileRouter } from "uploadthing/server";
+
+  type TRouter = FileRouter;
+  type TEndpoint = keyof TRouter;
+  type TSkipPolling = boolean;
+</script>
+
+<script
+  lang="ts"
+  generics="TRouter extends FileRouter , TEndpoint extends keyof TRouter, TSkipPolling extends boolean = false"
+>
+  import { onMount } from "svelte";
+  import { twMerge } from "tailwind-merge";
+
+  import {
+    allowedContentTextLabelGenerator,
+    resolveMaybeUrlArg,
+    styleFieldToClassName,
+  } from "@uploadthing/shared";
+  import type { StyleField } from "@uploadthing/shared";
+  import {
+    generateMimeTypes,
+    generatePermittedFileTypes,
+  } from "uploadthing/client";
+
+  import type { UploadthingComponentProps } from "../types";
+  import { INTERNAL_uploadthingHookGen } from "../useUploadThing";
+  import { getFilesFromClipboardEvent, progressWidths } from "./shared";
+  import Spinner from "./Spinner.svelte";
+
+  type ButtonStyleFieldCallbackArgs = {
+    __runtime: "svelte";
+    ready: boolean;
+    isUploading: boolean;
+    uploadProgress: number;
+    fileTypes: string[];
+  };
+
+  type UploadButtonAppearance = {
+    container?: StyleField<ButtonStyleFieldCallbackArgs>;
+    button?: StyleField<ButtonStyleFieldCallbackArgs>;
+    allowedContent?: StyleField<ButtonStyleFieldCallbackArgs>;
+    clearBtn?: StyleField<ButtonStyleFieldCallbackArgs>;
+  };
```

Comment:
I think we may need to adjust the types here to adjust for Svelte?

![CleanShot 2024-03-21 at 21 58 27@2x](https://github.com/pingdotgg/uploadthing/assets/51714798/17276618-cc1a-47b2-beb9-c144f75ac5b6)